### PR TITLE
fix api folder causing buggy behaviour

### PR DESCRIPTION
**Pull request by jgunnCO**
## Description

Due to what seems to be a bug with Nextjs 12.3.4 having identically named subfolders in both `/pages` and `/page/api` directories causes Nextjs to mistakenly treat the latter as pages, meaning the redirects from these API routes are not respected.

## Type of change

Please check the relevant options.

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [ ] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:

# Checklist:

- [ ] If I have listed dependencies above, I have ensured that they are present in the target branch.
- [ ] I have performed a self-review of my code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

### CODE REVIEW

**Review by dominicwest — APPROVED** [review]
Changes look fine but wondering if its worth nesting all next api pages under an additional api dir to avoid clashes?

File: packages/applicant/src/pages/mandatory-questions/[mandatoryQuestionId]/organisation-summary.page.tsx [hunk 1]
```diff
@@ -172,16 +172,18 @@ export default function MandatoryQuestionOrganisationSummaryPage({
...
+                  schemeId: mandatoryQuestion.schemeId.toString(),
+                  mandatoryQuestionId,
+                },
+              }}
             >
               <a
                 className="govuk-button"
                 data-module="govuk-button"
                 aria-disabled="false"
-                type="button"
```
Comment (dominicwest) [inline packages/applicant/src/pages/mandatory-questions/[mandatoryQuestionId]/organisation-summary.page.tsx:184]: Can re-add this - might be needed for accessibility stuff

File: packages/applicant/src/pages/mandatory-questions/[mandatoryQuestionId]/organisation-summary.page.tsx [hunk 2]
```diff
@@ -172,16 +172,18 @@ export default function MandatoryQuestionOrganisationSummaryPage({
             </dl>
 
             <Link
-              href={routes.api.mandatoryQuestions.createSubmission(
-                mandatoryQuestionId,
-                mandatoryQuestion.schemeId.toString()
-              )}
+              href={{
+                pathname: `/api/create-submission`,
```
Comment (dominicwest) [inline packages/applicant/src/pages/mandatory-questions/[mandatoryQuestionId]/organisation-summary.page.tsx:176]: I do hate the routes file but maybe we should be consistent and use it here

File: packages/applicant/src/pages/mandatory-questions/[mandatoryQuestionId]/organisation-summary.page.tsx (hunk 1 above)
Comment (jgunnCO) [reply packages/applicant/src/pages/mandatory-questions/[mandatoryQuestionId]/organisation-summary.page.tsx:184]: you're thinking of `role` - `type` is used to indicate the MIME type of the URL, so it clearly isn't being used correctly here and this won't do anything.

File: packages/applicant/src/pages/mandatory-questions/[mandatoryQuestionId]/organisation-summary.page.tsx (hunk 2 above)
Comment (jgunnCO) [reply packages/applicant/src/pages/mandatory-questions/[mandatoryQuestionId]/organisation-summary.page.tsx:176]: I genuinely think it's a bad idea and just makes the code harder to follow and read.

**Pull request merged**